### PR TITLE
Pandas update to 0.21.0

### LINKED ISSUE #1628: Master branch fails tests using Python 3.6.3
In response to the test failures in issue #1620, I ran the tests using Python 3.6.3 and received the same errors as the [travis run](https://travis-ci.org/open-source-economics/Tax-Calculator/jobs/298126321).  Further, I ran the tests on the master branch and received the same errors.  

```
================================================================================================================================== FAILURES ==================================================================================================================================
_____________________________________________________________________________________________________________________________ test_create_tables _____________________________________________________________________________________________________________________________

cps_subsample =         age_head  age_spouse      e00200p      e00900p      e02100p  \
445636        41           0   922,485.99      ...2  
52701       0    0        5  
66990       0    0       16  
428804      0    0        9  

[4565 rows x 54 columns]

    def test_create_tables(cps_subsample):
        # pylint: disable=too-many-statements
        # create a current-law Policy object and Calculator object calc1
        rec = Records.cps_constructor(data=cps_subsample)
        pol = Policy()
        calc1 = Calculator(policy=pol, records=rec)
        calc1.calc_all()
        # create a policy-reform Policy object and Calculator object calc2
        reform = {2013: {'_II_rt1': [0.15]}}
        pol.implement_reform(reform)
        calc2 = Calculator(policy=pol, records=rec)
        calc2.calc_all()
    
        # test creating various difference tables
    
        diff = create_difference_table(calc1.records, calc2.records,
                                       groupby='large_income_bins',
                                       income_measure='expanded_income',
                                       tax_to_diff='combined')
        assert isinstance(diff, pd.DataFrame)
        expected = [0.00,
                    0.01,
                    0.41,
                    0.84,
                    0.92,
                    1.10,
                    1.15,
                    1.04,
                    0.78,
                    0.27,
                    np.nan]
        assert np.allclose(diff['perc_aftertax'], expected,
                           atol=0.005, rtol=0.0, equal_nan=True)
    
        diff = create_difference_table(calc1.records, calc2.records,
                                       groupby='webapp_income_bins',
                                       income_measure='expanded_income',
                                       tax_to_diff='iitax')
        assert isinstance(diff, pd.DataFrame)
        expected = [0.00,
                    0.01,
                    0.41,
                    0.84,
                    0.92,
                    1.10,
                    1.15,
                    1.04,
                    0.78,
                    0.30,
                    0.08,
                    0.07,
                    np.nan]
        assert np.allclose(diff['perc_aftertax'], expected,
                           atol=0.005, rtol=0.0, equal_nan=True)
    
        diff = create_difference_table(calc1.records, calc2.records,
                                       groupby='small_income_bins',
                                       income_measure='expanded_income',
                                       tax_to_diff='iitax')
        assert isinstance(diff, pd.DataFrame)
        expected = [0.00,
                    0.01,
                    0.02,
                    0.16,
                    0.64,
                    0.82,
                    0.87,
                    0.92,
                    1.10,
                    1.15,
                    1.04,
                    0.78,
                    0.30,
                    0.08,
                    0.09,
                    0.07,
                    0.05,
                    0.02,
                    0.00,
                    np.nan]
>       assert np.allclose(diff['perc_aftertax'], expected,
                           atol=0.005, rtol=0.0, equal_nan=True)
E       assert False
E        +  where False = <function allclose at 0x7f98cb85a1e0>(0            0.00\n1            0.01\n2            0.02\n3            0.16\n4            0.64\n5            0.82\n6         ...       0.07\n16           0.05\n17           0.02\n18            nan\nsums          nan\nName: perc_aftertax, dtype: float64, [0.0, 0.01, 0.02, 0.16, 0.64, 0.82, ...], atol=0.005, rtol=0.0, equal_nan=True)
E        +    where <function allclose at 0x7f98cb85a1e0> = np.allclose

tests/test_utils.py:143: AssertionError
---------------------------------------------------------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------------------------------------------------------
You loaded data for 2014.
Tax-Calculator startup automatically extrapolated your data to 2014.
You loaded data for 2014.
Tax-Calculator startup automatically extrapolated your data to 2014.
________________________________________________________________________________________________________________________ test_weighted_count_lt_zero _________________________________________________________________________________________________________________________

    def test_weighted_count_lt_zero():
        df1 = pd.DataFrame(data=DATA, columns=['tax_diff', 's006', 'label'])
        grped = df1.groupby('label')
        diffs = grped.apply(weighted_count_lt_zero, 'tax_diff')
        exp = pd.Series(data=[4, 0], index=['a', 'b'])
        exp.index.name = 'label'
>       pd.util.testing.assert_series_equal(exp, diffs)

tests/test_utils.py:489: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../miniconda3/envs/taxcalc-dev/lib/python3.6/site-packages/pandas/util/testing.py:1238: in assert_series_equal
    assert_attr_equal('dtype', left, right)
../../miniconda3/envs/taxcalc-dev/lib/python3.6/site-packages/pandas/util/testing.py:1008: in assert_attr_equal
    raise_assert_detail(obj, msg, left_attr, right_attr)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

obj = 'Attributes', message = 'Attribute "dtype" are different', left = dtype('int64'), right = dtype('float64'), diff = None

    def raise_assert_detail(obj, message, left, right, diff=None):
        if isinstance(left, np.ndarray):
            left = pprint_thing(left)
        if isinstance(right, np.ndarray):
            right = pprint_thing(right)
    
        msg = """{obj} are different
    
    {message}
    [left]:  {left}
    [right]: {right}""".format(obj=obj, message=message, left=left, right=right)
    
        if diff is not None:
            msg += "\n[diff]: {diff}".format(diff=diff)
    
>       raise AssertionError(msg)
E       AssertionError: Attributes are different
E       
E       Attribute "dtype" are different
E       [left]:  int64
E       [right]: float64

../../miniconda3/envs/taxcalc-dev/lib/python3.6/site-packages/pandas/util/testing.py:1089: AssertionError
___________________________________________________________________________________________________________________________ test_weighted_perc_cut ___________________________________________________________________________________________________________________________

    def test_weighted_perc_cut():
        dfx = pd.DataFrame(data=DATA, columns=['tax_diff', 's006', 'label'])
        grouped = dfx.groupby('label')
        diffs = grouped.apply(weighted_perc_cut, 'tax_diff')
        exp = pd.Series(data=[4. / 12., 0.0], index=['a', 'b'])
        exp.index.name = 'label'
>       pd.util.testing.assert_series_equal(exp, diffs)

tests/test_utils.py:574: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../miniconda3/envs/taxcalc-dev/lib/python3.6/site-packages/pandas/util/testing.py:1272: in assert_series_equal
    obj='{obj}'.format(obj=obj))
pandas/_libs/testing.pyx:59: in pandas._libs.testing.assert_almost_equal
    ???
pandas/_libs/testing.pyx:173: in pandas._libs.testing.assert_almost_equal
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

obj = 'Series', message = 'Series values are different (50.0 %)', left = '[0.333333333333, 0.0]', right = '[0.333333333306, nan]', diff = None

    def raise_assert_detail(obj, message, left, right, diff=None):
        if isinstance(left, np.ndarray):
            left = pprint_thing(left)
        if isinstance(right, np.ndarray):
            right = pprint_thing(right)
    
        msg = """{obj} are different
    
    {message}
    [left]:  {left}
    [right]: {right}""".format(obj=obj, message=message, left=left, right=right)
    
        if diff is not None:
            msg += "\n[diff]: {diff}".format(diff=diff)
    
>       raise AssertionError(msg)
E       AssertionError: Series are different
E       
E       Series values are different (50.0 %)
E       [left]:  [0.333333333333, 0.0]
E       [right]: [0.333333333306, nan]

../../miniconda3/envs/taxcalc-dev/lib/python3.6/site-packages/pandas/util/testing.py:1089: AssertionError
__________________________________________________________________________________________________________________________ test_diff_table_sum_row ___________________________________________________________________________________________________________________________

cps_subsample =         age_head  age_spouse      e00200p      e00900p      e02100p  \
445636        41           0   922,485.99      ...2  
52701       0    0        5  
66990       0    0       16  
428804      0    0        9  

[4565 rows x 54 columns]

    def test_diff_table_sum_row(cps_subsample):
        rec = Records.cps_constructor(data=cps_subsample)
        # create a current-law Policy object and Calculator calc1
        pol = Policy()
        calc1 = Calculator(policy=pol, records=rec)
        calc1.calc_all()
        # create a policy-reform Policy object and Calculator calc2
        reform = {2013: {'_II_rt4': [0.56]}}
        pol.implement_reform(reform)
        calc2 = Calculator(policy=pol, records=rec)
        calc2.calc_all()
        # create two difference tables and compare their content
        tdiff1 = create_difference_table(calc1.records, calc2.records,
                                         groupby='small_income_bins',
                                         income_measure='expanded_income',
                                         tax_to_diff='iitax')
        tdiff2 = create_difference_table(calc1.records, calc2.records,
                                         groupby='large_income_bins',
                                         income_measure='expanded_income',
                                         tax_to_diff='iitax')
        non_digit_cols = ['mean', 'perc_inc', 'perc_cut', 'share_of_change',
                          'perc_aftertax', 'pc_aftertaxinc']
        digit_cols = [c for c in list(tdiff1) if c not in non_digit_cols]
>       assert np.allclose(tdiff1[digit_cols][-1:],
                           tdiff2[digit_cols][-1:])
E       assert False
E        +  where False = <function allclose at 0x7f98cb85a1e0>(            count    tax_cut    tax_inc     tot_change\nsums 1,641,299.36        nan  65,138.69 806,980,455.93,             count    tax_cut    tax_inc     tot_change\nsums 1,641,299.36        nan  65,138.69 806,980,455.93)
E        +    where <function allclose at 0x7f98cb85a1e0> = np.allclose

tests/test_utils.py:716: AssertionError
---------------------------------------------------------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------------------------------------------------------
You loaded data for 2014.
Tax-Calculator startup automatically extrapolated your data to 2014.
You loaded data for 2014.
Tax-Calculator startup automatically extrapolated your data to 2014.
============================================================================================================================ 68 tests deselected =============================================================================================================================

```

### PR DESCRIPTION
**Pull request by hdoupe**
Resolves #1628

### CODE REVIEW

File: taxcalc/tests/test_utils.py [hunk 1]
```diff
@@ -138,9 +138,9 @@ def test_create_tables(cps_subsample):
                 0.07,
                 0.05,
                 0.02,
-                0.00,
+                np.nan,
                 np.nan]
```
Comment (hdoupe) [inline taxcalc/tests/test_utils.py]: This needs to be reviewed further. @martinholmer @codykallen

**Comment by codykallen** [comment]
@hdoupe, do you know why that changed from 0.0 to nan? That income bin in for filing units with `AGI > 9999999`. Are there no filing units in the `cps_subsample` with AGI over $10 million?

File: taxcalc/tests/test_utils.py (hunk 1 above)
Comment (martinholmer) [reply taxcalc/tests/test_utils.py]: Why this change?

Comment (martinholmer) [reply taxcalc/tests/test_utils.py]: Why this change?

Comment (hdoupe) [reply taxcalc/tests/test_utils.py]: I suspect that it has something to do with [this example](https://pandas.pydata.org/pandas-docs/stable/basics.html#gotchas) from the Pandas docs.  However, I'm not sure why a why what should be a zero was converted to a NaN in our case.

**Comment by martinholmer** [comment]
@hdoupe,
Why the extra number in line 142?

**Comment by hdoupe** [comment]
@martinholmer said
>  Why the extra number in line 142?
 
I don't know what you mean.  I don't think that I added an extra number in `test_utils.py`.

**Comment by martinholmer** [comment]
You’re right. No extra number but a zero turning into NaN is a question

**Comment by hdoupe** [comment]
@codykallen That makes sense.  The errors in the other tests were caused by applying a function to an empty dataframe.

File: taxcalc/tests/test_utils.py (hunk 1 above)
Comment (hdoupe) [reply taxcalc/tests/test_utils.py]: @codykallen you may be correct.  The result of     `print('sum', np.sum(calc1.records.c00100 > 10000000) )` is `0`.

**Comment by codykallen** [comment]
@martinholmer, @MattHJensen 

If this PR looks good to you, can we merge it in, and start checking if the other PRs pass the new tests?

**Comment by MattHJensen** [comment]
@hdoupe, are you ready to remove the WIP tag?

**Comment by hdoupe** [comment]
@martinholmer I changed the `NaN` value back to `0`.  You made a good point when you said:
> Don’t change the tests!
> The code is wrong and needs to be changed

@codykallen @MattHJensen This PR should be ready to go.

**Comment by MattHJensen** [comment]
This looks good to me. Thanks very much @hdoupe. Contingent on the tests passing, I will merge it. 

As an aside, this PR demonstrates the only substantial downside of >= pinning (in my view), which is that you don't get to choose when to update packages, and you can be forced to deal with updates when you are in a hurry to deal with something else.

**Comment by hdoupe** [comment]
@MattHJensen said
> As an aside, this PR demonstrates the only substantial downside of >= pinning (in my view), which is that you don't get to choose when to update packages, and you can be forced to deal with updates when you are in a hurry to deal with something else.

I agree.  This time, we really were pretty busy.  However, sometimes I think that I'm too busy to deal with an annoying task like this, but I could actually find some time to get it done.  It can be difficult to figure out where that "too busy" line is.

**Pull request merged**